### PR TITLE
API-4299: sending teamMemberAddedEvent when collaborator added to an …

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
…application

### CODE REVIEW

File: test/unit/uk/gov/hmrc/thirdpartyapplication/services/ApplicationServiceSpec.scala [hunk 1]
```diff
@@ -117,6 +119,7 @@ class ApplicationServiceSpec extends AsyncHmrcSpec with BeforeAndAfterAll with A
     when(mockEmailConnector.sendApplicationApprovedAdminConfirmation(*, *, *)(*)).thenReturn(successful(response))
     when(mockEmailConnector.sendApplicationApprovedNotification(*, *)(*)).thenReturn(successful(response))
     when(mockEmailConnector.sendApplicationDeletedNotification(*, *, *)(*)).thenReturn(successful(response))
+    when(mockApiPlatformEventService.sendTeamMemberAddedEvent(any[ApplicationData], any[String], any[String])(any[HeaderCarrier])).thenReturn(successful(true))
```
Comment (lburgos) [inline test/unit/uk/gov/hmrc/thirdpartyapplication/services/ApplicationServiceSpec.scala:122]: Maybe we could add a test where we verify that the service sends this event when a collaborator is added.

File: app/uk/gov/hmrc/thirdpartyapplication/models/ApiPlatformEvents.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  val eventType: EventType.Value
+  val actor: Actor
+}
+
+case class TeamMemberAddedEvent(override val applicationId: String,
+                                override val actor: Actor,
+                                teamMemberEmail: String,
+                                teamMemberRole: String) extends ApplicationEvent {
+  override val eventType: EventType.Value = EventType.TEAM_MEMBER_ADDED
+  override val eventDateTime: DateTime = DateTime.now()
```
Comment (lburgos) [inline app/uk/gov/hmrc/thirdpartyapplication/models/ApiPlatformEvents.scala:41]: The eventDateTime won't be written to Json unless it's in the parameter list. The eventType will, as long as we use the Json union library.

File: app/uk/gov/hmrc/thirdpartyapplication/models/JsonFormatters.scala [hunk 3]
```diff
@@ -271,3 +271,8 @@ object APIStatusJson {
   }
 
 }
+
+object ApplicationEventFormats {
+  implicit val actorFormats = Json.format[Actor]
+  implicit val teamMemberAddedEventFormats = Json.format[TeamMemberAddedEvent]
```
Comment (lburgos) [inline app/uk/gov/hmrc/thirdpartyapplication/models/JsonFormatters.scala:277]: I think we need to use the json union library and create a formatter for the ApplicationEvent trait instead

Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/thirdpartyapplication/models/JsonFormatters.scala:277]: ok will add this

File: app/uk/gov/hmrc/thirdpartyapplication/models/ApiPlatformEvents.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/thirdpartyapplication/models/ApiPlatformEvents.scala:41]: ok will change this

File: test/unit/uk/gov/hmrc/thirdpartyapplication/services/ApplicationServiceSpec.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply test/unit/uk/gov/hmrc/thirdpartyapplication/services/ApplicationServiceSpec.scala:122]: yep will add a few verifies on the mock for adding a collaborator

**Pull request merged**